paddle/function/KernelType.h

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+#include <functional>
+#include "paddle/utils/Error.h"
+namespace paddle {
+class BufferArgs;
+namespace function {
+
+//! Kernel Function type of Paddle.
+//! Each layer will invoke this KernelType when forward/backward.
+typedef std::function<Error(BufferArgs& inputs, BufferArgs& outputs)>
+    KernelType;
```

Comment:
"kernel" is a polysemy. It could mean OS kernel, Linux kernel, CUDA kernel, etc. Here we need a specific name, e.g., `KernelType` ==> `CUDAKernel`.